### PR TITLE
Add `get_available_demos`

### LINKED ISSUE #1129: Add get_available_demos
### Problem Description

_As a user, it would be helpful to know the names of all available demos, as well as some information about each of those datasets._

### Acceptance criteria

- Add function to `demo.py` called `get_available_demos`
- Parameters:
    - (required) modality: One of 'single_table', 'multi_table' or 'sequential'
- Returns
    - A pandas.DataFrame containing the following columns for the specified modality:
        - `dataset_name`: The name of the dataset
        - `size_MB`: The size of the (unzipped) folder, in MB – Can be approximate. No need to download & save the actual data to verify the size.
        - num_tables: The number of tables in the dataset (this will be 1 if it's a single table or sequential modality) – To optimize: modality 'single_table' and 'sequential' can be hard-coded to 1. For 'multi_table', we can just count the # of .csv files in the folder; don't need to open up or parse the json.

### Expected behavior

```python
from sdv.datasets.demo import get_available_demos

get_available_demos(modality='single_table')

dataset_name            size_MB          num_tables
KRK_v1	                 0.321            1         
...

```

### Additional context

- Previously, we loaded the information to create the returned table from a csv stored in S3. We no longer want to do that. Instead, parse through the folder specified by the modality to get the information needed.
- Use the new S3 bucket

### PR DESCRIPTION
**Pull request by fealho**
Resolve #1129.

### CODE REVIEW

File: sdv/data_processing/data_processor.py [hunk 1]
```diff
@@ -498,7 +498,7 @@ def transform(self, data, is_condition=False):
 
         try:
             transformed = self._hyper_transformer.transform_subset(data)
-        except (rdt.errors.NotFittedError, rdt.errors.Error):
+        except (rdt.errors.NotFittedError, rdt.errors.ConfigNotSetError):
```
Comment (fealho) [inline sdv/data_processing/data_processor.py:501]: Due to RDT update.

File: sdv/datasets/demo.py [hunk 2]
```diff
@@ -125,10 +132,43 @@ def download_demo(modality, dataset_name, output_folder_name=None):
...
+                ``dataset_name``: The name of the dataset.
+                ``size_MB``: The unzipped folder size in MB.
+                ``num_tables``: The number of tables in the dataset.
+
+    Raises:
+        Error:
+            * If ``modality`` is not ``'single_table'``, ``'multi_table'`` or ``'sequential'``.
+    """
+    _validate_modalities(modality)
+    client = boto3.client('s3', config=Config(signature_version=UNSIGNED))
```
Comment (fealho) [inline sdv/datasets/demo.py:164]: As we discussed in my last meeting last year, I updated the previous logic (which used `requests` and `ET`) to use boto.

File: sdv/multi_table/hma.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ def _pop_foreign_keys(self, table_data, table_name):
     def _clear_nans(table_data):
         for column in table_data.columns:
             column_data = table_data[column]
-            if column_data.dtype in (np.int, np.float):
+            if column_data.dtype in (int, float):
```
Comment (fealho) [inline sdv/multi_table/hma.py:157]: Due to numpy 1.24 release.

**Review by amontanez24 — COMMENTED** [review]
This looks really good! One small comment about the tests but after that I'd say it's good to go

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Review by amontanez24 — APPROVED** [review]
Thanks for addressing! :shipit:

**Pull request merged**